src/xcode/ENA/ENA/Source/Scenes/Home/HomeCoordinator.swift

```diff
@@ -159,13 +159,13 @@ class HomeCoordinator: RequiresAppDependencies {
 		})
 	}
 
-	func showTestResultFromNotification(with testType: CoronaTestType) {
+	func showTestResultFromNotification(with testType: CoronaTestType, visibleViewController: UIViewController?) {
 		if let presentedViewController = rootViewController.presentedViewController {
 			presentedViewController.dismiss(animated: true) {
-				self.showExposureSubmission(testType: testType)
+				self.showExposureSubmission(testType: testType, visibleViewController: visibleViewController)
```

Comment:
Not sure if I agree with this approach to just present it on some other view controller, because this could even be the exposure submission flow already, couldn't it? So we would present it twice? And in line 164 we are already trying to dismiss any view controllers that are already presented, so the former "visible view controller" should actually be dismissed right before you are trying to present from it. So I'm wondering why this works at all and why it didn't work as expected before 😅